batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -12,22 +16,38 @@ use crate::{
     core::{
         errors::SubmitError,
         types::{
-            AlignedVerificationData, ClientMessage, ResponseMessage, VerificationData,
-            VerificationDataCommitment,
+            AlignedVerificationData, ClientMessage, NoncedVerificationData, ResponseMessage,
+            VerificationData, VerificationDataCommitment,
         },
     },
+    eth::batcher_payment_service::BatcherPaymentService,
 };
 
 pub async fn send_messages(
     ws_write: Arc<Mutex<SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, Message>>>,
     verification_data: &[VerificationData],
     wallet: Wallet<SigningKey>,
-) -> Result<Vec<VerificationData>, SubmitError> {
+    batcher_payment_service: &BatcherPaymentService,
+) -> Result<Vec<NoncedVerificationData>, SubmitError> {
     let mut sent_verification_data = Vec::new();
 
     let mut ws_write = ws_write.lock().await;
 
+    let address = wallet.address();
+
+    let mut nonce = batcher_payment_service
+        .user_nonces(address)
+        .await
+        .map_err(|_| SubmitError::GenericError("Invalid Nonce".to_string()))?;
```

Comment:
This may not work if the user sends to transaction before the nonce gets updated.